cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperator.java

```diff
@@ -626,6 +632,15 @@ public void checkCustomCaSecret(CertificateAuthority ca, Secret certSecret, Secr
                                 return Future.succeededFuture();
                             }
                         })
+                        .compose(i -> deploymentOperations.getAsync(namespace, KafkaExporter.kafkaExporterName(name)))
+                        .compose(dep -> {
+                            if (dep != null) {
+                                log.debug("{}: Rolling Deployment {} to {}", reconciliation, KafkaExporter.kafkaExporterName(name), reasons);
+                                return deploymentOperations.rollingUpdate(namespace, KafkaExporter.kafkaExporterName(name), operationTimeoutMs);
+                            } else {
+                                return Future.succeededFuture();
+                            }
+                        })
```

Comment:
Not for this PR, but I look at these three pairs of compose operations on deployments of different names doing an async get and a roll and I think we should factor out a common method. It's just the name of the deployment which seems to differ.

Reply:
I do not think this is generaly usable to add it to the resource operators. But I factored it out into a searate method inside the `KafkaAssemblyOperator` it self.